### PR TITLE
[sovereign] RevertStateToBlock and save block data to storage

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Several extra implementation for sovereign epoch change flow
  
## Proposed changes
- `RevertStateToBlock` for `sovereignTrigger`
- `SaveBlockDataToStorage` for `epochRewardsCreator` at end of epoch
- Assign `scbp.blockProcessor = scbp` so that we can call `removeStartOfEpochBlockDataFromPools` for sovereign

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/block/sovereignChainBlock.go [hunk 1]
```diff
@@ -1466,9 +1466,13 @@ func (scbp *sovereignChainBlockProcessor) commitEpochStart(header data.HeaderHan
...
 		scbp.createEpochStartData(body)
 		go scbp.validatorInfoCreator.SaveBlockDataToStorage(header, body)
-		// TODO: MX-15588 FIX THIS (mp *metaProcessor) commitEpochStart
-		//go scbp.epochRewardsCreator.SaveBlockDataToStorage(header, body)
-		// ************
+
+		sovMetaHdr, castOk := header.(data.MetaHeaderHandler)
+		if !castOk {
+			log.Error("sovereignChainBlockProcessor.commitEpochStart", "error", process.ErrWrongTypeAssertion)
+		}
```
Comment (sasurobert) [inline process/block/sovereignChainBlock.go:1473]: you missed a return here.

Comment (sasurobert) [reply process/block/sovereignChainBlock.go:1473]: also this should be a critical error.

File: epochStart/metachain/trigger.go [hunk 2]
```diff
@@ -386,7 +385,12 @@ func (t *trigger) revert(header data.HeaderHandler) error {
 		return err
 	}
 
-	epochStartIdentifier := core.EpochStartIdentifier(metaHdr.Epoch)
+	t.baeRevert(epochStartMeta, metaHdr)
```
Comment (sasurobert) [inline epochStart/metachain/trigger.go:388]: bae?

File: epochStart/metachain/sovereignTrigger.go [hunk 3]
```diff
@@ -36,6 +41,77 @@ func (st *sovereignTrigger) SetProcessed(header data.HeaderHandler, body data.Bo
 	st.baseSetProcessed(sovChainHeader, body)
 }
 
+// RevertStateToBlock will revert the state of the trigger to the current block
+func (st *sovereignTrigger) RevertStateToBlock(header data.HeaderHandler) error {
```
Comment (axenteoctavian) [inline epochStart/metachain/sovereignTrigger.go:45]: I suggest you can add `sovereignTrigger.RevertStateToBlock` in all 3 log messages in this func

File: process/block/sovereignChainBlock.go (hunk 1 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:1488]: Right, missed it, added it

File: epochStart/metachain/sovereignTrigger.go (hunk 3 above)
Comment (mariusmihaic) [reply epochStart/metachain/sovereignTrigger.go:45]: Added it

**Review by sasurobert — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]

**Pull request merged**